virttest/utils_numeric.py

```diff
@@ -37,3 +40,38 @@ def format_size_human_readable(value, binary=False, precision='%.2f'):
     value = value * base / unit
     format_str = ('%d' if value.is_integer() else precision) + ' %s'
     return format_str % (value, s)
+
+
+def normalize_data_size(value_str, order_magnitude="M", factor="1024"):
+    """
+    Normalize a data size in one order of magnitude to another (MB to GB,
+    for example).
+
+    :param value_str: a string include the data default unit is 'B'
+    :param order_magnitude: the magnitude order of result
+    :param factor: the factor between two relative order of magnitude.
+                   Normally could be 1024 or 1000
+    """
+    def __get_unit_index(M):
+        try:
+            return ['B', 'K', 'M', 'G', 'T'].index(M.upper())
+        except ValueError:
+            pass
+        return 0
+
+    regex = r"(\d+\.?\d*)\s*(\w?)"
+    match = re.search(regex, value_str)
+    try:
+        value = match.group(1)
+        unit = match.group(2)
+        if not unit:
+            unit = 'B'
+    except TypeError:
+        raise ValueError("Invalid data size format 'value_str=%s'" % value_str)
+    from_index = __get_unit_index(unit)
+    to_index = __get_unit_index(order_magnitude)
+    scale = int(factor) ** (to_index - from_index)
```

Comment:
`scale = Decimal(factor) ** (to_index - from_index)` will solve the precision error.

Reply:
>>> from virttest import utils_numeric
>>> utils_numeric.normalize_data_size('1M', 'K', '1000')
'1E+3'
>>> 
From the above result , may be it's not a friendly readable format.

Reply:
could disable `Decimal` scientific  representation just like:
`return format(data_size, 'f')`
float type has the same implicit scientific problem, so it is better to add extra parameter to let user to pass a format string like '%.2f', using format string could easily convert into string without scientific notation.